pkg/plugins/git/generic/main_test.go

```diff
@@ -1,11 +1,72 @@
 package generic
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
 )
 
+func TestIsSimilarBranch(t *testing.T) {
+
+	type data struct {
+		branchA        string
+		branchB        string
+		workingDir     string
+		expectedResult bool
+		expectedError  error
+	}
+
+	type dataSet []data
+
+	dSet := dataSet{
+		{
+			branchA:        "main",
```

Comment:
This test assumes that the project directory hosting the code has a local git reference (short) named `main`.

- It isn't verified in CI systems (GH Actions, but alos Gitlab and Jenkins and many others): only the current HEAD's reference (e.g. the PR source branch) is checkouted (to avoid heavy checkouts all the time). In the case of this PR it's even worse: the local HEAD's reference is not deterministic because it's a merge of the HEAD reference with the destination's branch.
- It isn't verified on contributor machines: as the contributing rules says, a topic/feature branch is needed to open a PR: there is no reason to have checkouted the `main` before running the test.


This test is not valid because it relies on a state that you do not control.

WDYT about creating (and initializing) an empty git workdir for each dataSet, so you control the git references (this is an integration test because it interacts with a real git project)?

The alternative is to make it a unit test, but it needs more heavy lifting (as you would have to mock the git project).

Reply:
> It isn't verified in CI systems (GH Actions, but alos Gitlab and Jenkins and many others): only the current HEAD's reference (e.g. the PR source branch) is checkouted (to avoid heavy checkouts all the time). In the case of this PR it's even worse: the local HEAD's reference is not deterministic because it's a merge of the HEAD reference with the destination's branch.

Thanks, that was the purpose of this [commit](https://github.com/updatecli/updatecli/pull/436/commits/bf6067c4d33fbf7a9f8789db82e1a54cf1cdfadb), I thought that I could rely on the branches but it doesn'
t seem so, I'll comment my test.

> WDYT about creating (and initializing) an empty git workdir for each dataSet, so you control the git references (this is an integration test because it interacts with a real git project)?
The alternative is to make it a unit test, but it needs more heavy lifting (as you would have to mock the git project).

I prefer the UT approach but I won't have the time to work on this now and I think the fix is more urgent. so I'll comment my tests for now